truss/main.go

```diff
@@ -145,40 +221,40 @@ func mkdir(path string) error {
 	return err
 }
 
+// exitIfError will print the error message and exit 1 if the passed error is
+// non-nil
 func exitIfError(err error) {
 	if errors.Cause(err) != nil {
 		defer os.Exit(1)
 		fmt.Printf("%v\n", err)
 	}
 }
 
-// readPreviousGeneration accepts the path to the directory where the inputed .proto files are stored, protoDir,
-// it returns a []truss.NamedReadWriter for all files in the service/ dir in protoDir
-func readPreviousGeneration(protoDir, packageName string) ([]truss.NamedReadWriter, error) {
-	dir := protoDir + "/" + packageName + "-service"
-	if fileExists(dir) != true {
+// readPreviousGeneration returns a []truss.NamedReadWriter for all files serviceDir
+func readPreviousGeneration(serviceDir string) ([]truss.NamedReadWriter, error) {
+	if fileExists(serviceDir) != true {
 		return nil, nil
 	}
 
 	var files []truss.NamedReadWriter
 	sfs := simpleFileConstructor{
-		protoDir: protoDir,
-		files:    files,
+		dirPath: filepath.Dir(serviceDir),
```

Comment:
That definition makes sense; I can't argue with it. And by it, `dirPath` makes sense. It just sounds weird. I understand not calling it `path`. I probably would have just called it `dir`. But you've provided an adequate defense, if you want to keep it `dirPath`. It just makes my eyebrows twist a little to see it. :)